src/content/peerconnection/endtoend-encryption/js/videopipe.js

```diff
@@ -0,0 +1,102 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+//
+// A "videopipe" abstraction on top of WebRTC.
+//
+// The usage of this abstraction:
+// var pipe = new VideoPipe(mediastream, handlerFunction);
+// handlerFunction = function(mediastream) {
+//   do_something
+// }
+// pipe.close();
+//
+// The VideoPipe will set up 2 PeerConnections, connect them to each
+// other, and call HandlerFunction when the stream is available in the
+// second PeerConnection.
+//
+'use strict';
+
+function errorHandler(context) {
+  return function(error) {
+    trace('Failure in ' + context + ': ' + error.toString);
+  };
+}
+
+// eslint-disable-next-line no-unused-vars
+function successHandler(context) {
+  return function() {
+    trace('Success in ' + context);
+  };
+}
+
+function noAction() {
+}
+
+
+function VideoPipe(stream, sendTransform, receiveTransform, handler) {
+  const pc1 = new RTCPeerConnection({
+    forceEncodedVideoInsertableStreams: !!sendTransform
+  });
+  const pc2 = new RTCPeerConnection({
+    forceEncodedVideoInsertableStreams: !!receiveTransform
+  });
+
+  const sender = pc1.addTrack(stream.getVideoTracks()[0], stream);
+  if (sendTransform) {
+    const senderStreams = sender.createEncodedVideoStreams();
+    const senderTransformStream = new TransformStream({
+      start() {},
+      flush() {},
+      transform: sendTransform
+    });
+    senderStreams.readableStream
```

Comment:
would it be interesting to do the encrypt/decrypt on a worker thread by transfering the streams to it?

Reply:
Yep. That's a later demo.
This doesn't work yet (the CL for making frames transferable is still in review), so I didn't include it at this time.